src/utils/line.ts

```diff
@@ -51,54 +58,138 @@ export const formatCreditEvents = (
   price: BigNumber = BigNumber.from(0),
   events: LineEventFragResponse[]
 ): CreditEvent[] => {
-  return events.map((e: any): CreditEvent => {
-    const { __typename, amount, token, credit, timestamp, value = unnullify(0, true) } = e;
+  return events.map((event: any): CreditEvent => {
+    const { id, timestamp, __typename, amount, value, position } = event;
     return {
-      id: credit?.id,
+      id,
       __typename,
       timestamp,
       amount,
+      // value: unnullify(value, true),
       value,
-      token: token?.id,
-      currentValue: price.mul(value),
+      token: position.token.id,
+      // currentValue: price.mul(value),
     };
   });
 };
 
+// TODO: Modify/adjust this to replace mergeCollateralEvents (formatCollateralEvents)
+// given that there are 3 duplicative uses of collateral events in this file!!
+export const formatEscrowToCollateralEvents = (escrow: AggregatedEscrow | undefined): CollateralEvent[] => {
+  // console.log('get line events escrow: ', escrow);
+  const collateralEvents: CollateralEvent[] = escrow!
```

Comment:
if u know its going to exist then dont make it undefined in the parameter (can also do `?: AgreegatedEscrow` as shorthand for possible undefined

Reply:
Thanks for the tip!

Doesn't `?: AgreegatedEscrow` mean it's an optional parameter as opposed to a required parameter than it of type `AggregatedEscrow` or `undefined`? 

Not sure if there is any logical difference between the two scenarios I've described.